app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java

```diff
@@ -304,21 +306,24 @@ public void done(com.parse.ParseException e) {
      * @param md marker detail
      */
     public static Marker createMarker(GoogleMap mMap, MarkerDetails md) throws com.parse.ParseException {
-        Post post = (Post) md.getPost();
-        double latitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLatitude();
-        double longitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLongitude();
-        String title = (post.fetchIfNeeded().getString("title"));
-        ParseFile image = post.fetchIfNeeded().getParseFile("image");
-        String imageUrl = "";
-        if (image != null) {
-            imageUrl = image.getUrl();
+        if(md.getPost()!=null) {
+            Post post = (Post) md.getPost();
+            double latitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLatitude();
```

Comment:
Make "location" a constant. It also might be easier here to do ->
  Latlng location = post.fetchIfNeeded().getParseGeoPoint("location");
and just use `location.getlatitude()` and `location.getLongitude()` in your code when needed